tests/tests_pytorch/loggers/test_mlflow.py

```diff
@@ -270,6 +257,38 @@ def test_mlflow_logger_experiment_calls(client, _, time, param, metric, tmpdir):
     )
 
 
+def _check_value_length(value, *args, **kwargs):
+    assert len(value) <= 250
+
+
+@mock.patch("pytorch_lightning.loggers.mlflow.Param", side_effect=_check_value_length)
+@mock.patch("pytorch_lightning.loggers.mlflow._MLFLOW_AVAILABLE", return_value=True)
+@mock.patch("pytorch_lightning.loggers.mlflow.MlflowClient")
+def test_mlflow_logger_with_long_param_value(client, _, param, tmpdir):
+    """Test that long parameter values are truncated to 250 characters."""
+    logger = MLFlowLogger("test", save_dir=tmpdir)
+
+    params = {"test": "test_param" * 50}
+    logger.log_hyperparams(params)
+
+    # assert_called_once_with() won't properly check the parameter value.
+    logger.experiment.log_batch.assert_called_once()
```

Comment:
yes, i found that too.
you can see in the test above `test_mlflow_logger_experiment_calls` how I did it:

```py
    logger.experiment.log_batch.assert_called_once_with(
        run_id=logger.run_id, params=[param(key="test", value="test_param")]
    )
    param.assert_called_with(key="test", value="test_param")
```

Reply:
@awaelchli does that work correctly, though? Isn't `param` a different instance of `MagicMock` (instantiated on the previous line) than [the one instantiated in `log_hyperparams()`](https://github.com/Lightning-AI/lightning/blob/master/src/pytorch_lightning/loggers/mlflow.py#L250)?